### PR TITLE
resolves #15 exclude pages marked as 'noindex'

### LINKED ISSUE #15: Allow pages to be excluded from indexing
The indexer should skip pages which are marked as "noindex". For example, consider a page which has the following meta tag:

```html
<meta name="robots" content="noindex">
```

Another option is to honor exclusions in robot.txt (or perhaps it should be both).

Here's what Google has to say about noindex: https://support.google.com/webmasters/answer/93710?hl=en

Here are some other ideas around inclusion/exclusion to consider: https://swiftype.com/documentation/site-search/crawler-configuration/content-inclusion-exclusion

### PR DESCRIPTION
**Pull request by Mogztter**
resolves #15

### CODE REVIEW

**Comment by mojavelinux** [comment]
Can we support either the meta tag or the page attribute? That seems like it would give us the most flexibility (given how the page is created).

File: lib/generate-index.js [hunk 1]
```diff
@@ -28,52 +28,60 @@ function generateIndex (playbook, pages) {
...
+  const documents = pages
+    .map((page) => {
+      const html = page.contents.toString()
+      const $ = cheerio.load(html)
+      return { page, $ }
+    })
+    // Exclude pages marked as "noindex"
+    .filter(({ page, $ }) => {
+      const $metaRobots = $('meta[name=robots]')
+      return !(($metaRobots && $metaRobots.attr('content') === 'noindex') || (page.asciidoc && page.asciidoc.noindex === ''))
```
Comment (Mogztter) [inline lib/generate-index.js]: @mojavelinux is it the correct way to check if an attribute is defined?

```adoc
= Secret page
:noindex:

This page won't be indexed.
```
```rb
page.asciidoc && page.asciidoc.noindex === '' # trueish
```

Comment (Mogztter) [reply lib/generate-index.js]: @mojavelinux Could you please confirm that this is the right way to check if `noindex` attribute is defined?

Apart from that I think this pull request is ready to be merged.

Comment (mojavelinux) [reply lib/generate-index.js]: It should be `page.asciidoc.attributes.noindex`. Otherwise, yes, you have the right idea.

**Comment by Mogztter** [comment]
Thanks @mojavelinux for your review 👍

**Pull request merged**